client_test.go

```diff
@@ -1,47 +1,184 @@
+		}
+
+		Convey("When cache attempts to store session", func() {
+			err := mockClient.Set(s)
 
+			Convey("Then session will not be stored", func() {
+				So(mockRedisClient.SetCalls(), ShouldHaveLength, 1)
+				So(err, ShouldNotBeEmpty)
+				So(err.Error(), ShouldEqual, "failed to store session")
+			})
+		})
 	})
 
-	Convey("Given redis client is set up", t, func() {
+	Convey("Given an invalid session", t, func() {
+		mockRedisClient, mockClient := setUpMocks(*redis.NewStatusCmd(), *redis.NewStringCmd())
 
-		mockClient := mock.ClientManagerMock{SetFunc: func(s *session.Session) error {
-			return errors.New("empty session provided")
-		}}
+		var s *session.Session = nil
 
-		Convey("Where the session is empty", func() {
-			var s *session.Session = nil
+		Convey("When cache attempts to store session", func() {
 			err := mockClient.Set(s)
 
-			Convey("Then session is not stored in redis", func() {
-				So(mockClient.SetCalls(), ShouldHaveLength, 1)
+			Convey("Then session will not be stored", func() {
+				So(mockRedisClient.SetCalls(), ShouldHaveLength, 0)
+				So(err, ShouldNotBeEmpty)
+				So(err.Error(), ShouldEqual, "session is empty")
+			})
+		})
+	})
+}
+
+func TestClient_GetByID(t *testing.T) {
+	Convey("Given a session ID", t, func() {
+		mockRedisClient, mockClient := setUpMocks(*redis.NewStatusCmd(), *redis.NewStringResult(string(resp), nil))
+
+		Convey("When cache attempts to get session", func() {
+			s, err := mockClient.GetByID("1234")
+			So(err, ShouldBeNil)
+
+			msg, err := mockRedisClient.Get(s.ID).Result()
+			So(err, ShouldBeNil)
+
+			var storedSession session.Session
+			err = json.Unmarshal([]byte(msg), &storedSession)
+			So(err, ShouldBeNil)
+
+			Convey("Then session is returned", func() {
+				So(storedSession.ID, ShouldEqual, s.ID)
+				So(storedSession.Email, ShouldEqual, s.Email)
+			})
+		})
+	})
+
+	Convey("Given a blank session ID", t, func() {
+		_, mockClient := setUpMocks(*redis.NewStatusCmd(), *redis.NewStringCmd())
+
+		Convey("When cache attempts to get session", func() {
+			_, err := mockClient.GetByID("")
+
+			Convey("Then session is returned", func() {
 				So(err, ShouldNotBeEmpty)
+				So(err.Error(), ShouldEqual, "id value is blank")
 			})
 		})
 	})
+
+	Convey("Given a session ID", t, func() {
+		mockRedisClient, _ := setUpMocks(*redis.NewStatusCmd(), *redis.NewStringResult("", errors.New("")))
+
+		Convey("When cache attempts to get session", func() {
+			msg, err := mockRedisClient.Get("1234").Result()
+
+			var storedSession session.Session
+			err = json.Unmarshal([]byte(msg), &storedSession)
+
+			Convey("Then session is returned", func() {
+				So(err, ShouldNotBeEmpty)
+				So(err.Error(), ShouldEqual, "unexpected end of JSON input")
+			})
+		})
+	})
+}
+
+func setUpClient(addr, password string, database int, ttl time.Duration) (*Client, error) {
+	c, err := NewClient(Config{
+		Addr:     addr,
+		Password: password,
+		Database: database,
+		TTL:      ttl,
+	})
+	return c, err
```

Comment:
This method isn't adding any functionality - remove it and use the `NewClient` function directly.

Reply:
Agreed, no idea why I did this! Removed.